### PR TITLE
Add MaterialTag.has_eye property

### PR DESCRIPTION
**Pull request by tal5**
Adds a MaterialTag property that controls whether end portal frames have an ender eye.

### CODE REVIEW

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/PropertyRegistry.java [hunk 1]
```diff
@@ -220,6 +220,7 @@ public static void registerMainProperties() {
         PropertyParser.registerProperty(MaterialSwitchFace.class, MaterialTag.class);
         PropertyParser.registerProperty(MaterialUnstable.class, MaterialTag.class);
         PropertyParser.registerProperty(MaterialWaterlogged.class, MaterialTag.class);
+        PropertyParser.registerProperty(MaterialHasEye.class, MaterialTag.class);
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/PropertyRegistry.java]: out of alphabetical order

**Comment by mcmonkey4eva** [comment]
Is there any reason this is its own property and not just bulked into `switched`?

**Comment by tal5** [comment]
> Is there any reason this is its own property and not just bulked into `switched`?

Added into `MaterialTag.switched`

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialSwitchable.java [hunk 2]
```diff
@@ -128,8 +139,10 @@ else if (isDaylightDetector()) {
         else if (isLightable()) {
             return getLightable().isLit();
         }
-        else {
+        else if (isPiston()) {
             return getPiston().isExtended();
+        } else {
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialSwitchable.java]: basic java formatting is wrong

Comment (tal5) [reply plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialSwitchable.java]: Missed that, my bad

**Pull request merged**